### PR TITLE
fix livecheck of bilibili

### PR DESCRIPTION
**Pull request by Aaron-212**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---

Related to #171006

### CODE REVIEW

File: Casks/b/bilibili.rb [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
       verified: "dl.hdslb.com/"
   name "Bilibili"
   name "哔哩哔哩官方客户端"
   desc "Official bilibili video streaming and sharing platform"
   homepage "https://app.bilibili.com/"
 
   livecheck do
-    url :url
-    strategy :extract_plist
+    url "http://api.bilibili.com/x/elec-frontend/update/latest.yml"
```
Comment (krehel) [inline Casks/b/bilibili.rb:13]: Let's use the secure connection if available
```suggestion
    url "https://api.bilibili.com/x/elec-frontend/update/latest.yml"
```

Comment (krehel) [inline Casks/b/bilibili.rb:13]: This doesn't look to contain the macOS versions - I see only .exe files.  Are the versions aligned with Windows and Mac always?  This seems like something that may break in the future.

Comment (Bo98) [reply Casks/b/bilibili.rb:13]: Probably should be `latest-mac.yml`

File: Casks/b/bilibili.rb [hunk 2]
```diff
@@ -1,17 +1,17 @@
 cask "bilibili" do
-  version "1.13.5.3377"
-  sha256 :no_check
+  version "1.13.5"
+  sha256 "c3f1f43bfc8836cb27d8acbff24834129366a4e30920c3113cccb5f9e70509ac"
 
-  url "https://dl.hdslb.com/mobile/fixed/pc_electron_mac/bili_mac.dmg",
+  url "https://dl.hdslb.com/mobile/fixed/pc_electron_mac/bili_mac.dmg?v=#{version}",
```
Comment (Bo98) [inline Casks/b/bilibili.rb:5]: Are you sure this works? I tried `https://dl.hdslb.com/mobile/fixed/pc_electron_mac/bili_mac.dmg?v=1.13.4` and still got 1.13.5.

File: Casks/b/bilibili.rb (hunk 1 above)
Comment (krehel) [reply Casks/b/bilibili.rb:13]: Yeah fair point that

File: Casks/b/bilibili.rb (hunk 2 above)
Comment (Bo98) [reply Casks/b/bilibili.rb:5]: `https://api.bilibili.com/x/elec-frontend/update/bilibili-v1.13.5.dmg` does work but only if passed a `appversion: 0.0.0` header (but does a 302 redirect to a URL that varies per version but doesn't need a header)

Comment (Aaron-212) [reply Casks/b/bilibili.rb]: Got the following

```xml
<?xml version="1.0" encoding="UTF-8"?>
<Error>
  <Code>NoSuchKey</Code>
  <Message>The specified key does not exist.</Message>
  <RequestId>665AA1B29C3A07333610694F</RequestId>
  <HostId>dl.hdslb.com</HostId>
  <Key>mobile/pack/bili_win/14880879/public/bilibili-v1.13.4.dmg</Key>
  <EC>0026-00000001</EC>
  <RecommendDoc>https://api.aliyun.com/troubleshoot?q=0026-00000001</RecommendDoc>
</Error>
```

So I think we should still use the unversioned url

**Review by miccal — APPROVED** [review]

**Pull request merged**